server/routes.go

```diff
@@ -1448,3 +1459,54 @@ func handleScheduleError(c *gin.Context, name string, err error) {
 		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
 	}
 }
+
+// prometheusMetricsMiddleware returns a Gin middleware handler that collects OpenTelemetry metrics for HTTP requests.
+// It records the route, response status code, and response status text for each request.
+func prometheusMetricsMiddleware(m *telemetry.Metrics) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		route := c.Request.URL.Path
+		c.Next()
+
+		// access the status we are sending
+		responseStatus := c.Writer.Status()
+		statusText := http.StatusText(responseStatus)
+
+		// record the request metric
+		m.RecordRequests(c.Request.Context(), "all", int64(responseStatus), statusText)
+
+		// record the metric based on request path
+		switch route {
+		case "/api/pull":
```

Comment:
This get's easily out of sync with the existing api. Some generic middleware  would make sense instead. I am not very familiar with gin but have you looked into https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation/github.com/gin-gonic/gin/otelgin for some inspiration?

Reply:
@alpe Usually this is the way you would increase the counters within the context for Prometheus metrics. While OpenTelemetry is one thing and I believe this is what you are referring to, for Prometheus this is absolutely enough.

Reply:
Updated it to set the action based on path by capturing request path.